packages/appeals-service-api/src/routes/v2/users/controller.js

```diff
@@ -1,76 +1,110 @@
-const { getUserByEmail, createUser, linkUserToAppeal } = require('./service');
-const logger = require('#lib/logger');
+const {
+	createUser,
+	searchUsers,
+	getUserByEmail,
+	getUserById,
+	updateUser,
+	removeLPAUser,
+	linkUserToAppeal
+} = require('./service');
 const ApiError = require('#errors/apiError');
 
 /**
- * @type {import('express').Handler}
+ * @type {import('express').RequestHandler}
  */
-async function userGet(req, res) {
-	let statusCode = 200;
-	let body = {};
-
-	try {
-		body = await getUserByEmail(req.params.email);
-	} catch (error) {
-		if (error instanceof ApiError) {
-			logger.error(`Failed to get users: ${error.code} // ${error.message.errors}`);
-			statusCode = error.code;
-			body = error.message.errors;
-		} else {
-			logger.error('Error:', error);
-			statusCode = 500;
-			body = 'An unexpected error occurred';
-		}
-	} finally {
-		res.status(statusCode).send(body);
+async function userPost(req, res) {
+	if ('id' in req.body) {
+		throw ApiError.badRequest({ errors: ['id is not allowed'] });
 	}
+	const body = await createUser(req.body);
+	res.status(200).send(body);
 }
 
 /**
- * @type {import('express').Handler}
+ * @type {import('express').RequestHandler}
  */
-async function userPost(req, res) {
-	if ('id' in req.body) {
-		throw ApiError.badRequest({ errors: ['id is not allowed'] });
-	}
-	let statusCode = 200;
-	let body = {};
-
-	try {
-		body = await createUser(req.body);
-	} catch (error) {
-		if (error instanceof ApiError) {
-			logger.error(`Failed create: ${error.code} // ${error.message.errors}`);
-			statusCode = error.code;
-			body = error.message.errors;
-		} else {
-			logger.error('Error:', error);
-			statusCode = 500;
-			body = 'An unexpected error occurred';
-		}
-	} finally {
-		res.status(statusCode).send(body);
-	}
+async function userSearch(req, res) {
+	const { lpaCode } = req.query;
+
+	const body = await searchUsers({
+		lpaCode: lpaCode?.trim()
+	});
+
+	res.status(200).send(body);
+}
+
+/**
+ * @type {import('express').RequestHandler}
+ */
+async function userGet(req, res) {
+	const body = await resolveUser(req.params.userLookup);
+	res.status(200).send(body);
+}
+
+/**
+ * @type {import('express').RequestHandler}
+ */
+async function userUpdate(req, res) {
+	const user = await resolveUser(req.params.userLookup);
+
+	const { isEnrolled, lpaStatus } = req.body;
+	if (isEnrolled) user.isEnrolled = isEnrolled;
+	if (lpaStatus) user.lpaStatus = lpaStatus;
```

Comment:
Do we need to check for `!== undefined` rather than truthy?

Reply:
So I added the open api format checker that validates the type with 400
lpaStatus is a string so think that should be fine

isEnrolled is boolean, but I think this should only work 1 way, i.e. can't unenroll

Reply:
Will add a comment